src/library/scala/jdk/package.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala
+
+/** The jdk package contains utilities to interact with JDK classes.
+ * 
+ * This packages offers a number of converters, that are able to wrap or copy
+ * types from the scala library to equivalent types in the JDK class library
+ * and vice versa:
+ * 
+ *  - [[CollectionConverters]], converting collections like [[scala.collection.Seq]],
+ *    [[scala.collection.Map]], [[scala.collection.Set]],
+ *    [[scala.collection.mutable.Buffer]], [[scala.collection.Iterator]]
+ *    and [[scala.collection.Iterable]] to their JDK counterparts
+ *  - [[OptionConverters]], converting between [[Option]] and
+ *    [[java.util.Optional]] and primitive variations
+ *  - [[StreamConverters]], to create JDK Streams from scala collections
+ *  - [[DurationConverters]], for conversions between scala
+ *    [[scala.concurrent.duration.FiniteDuration]] and [[java.time.Duration]]
+ *  - [[FunctionConverters]], from scala Functions to java
+ *    [[java.util.function.Function]], [[java.util.function.UnaryOperator]],
+ *    [[java.util.function.Consumer]] and [[java.util.function.Predicate]], as
+ *    well as primitive variations and Bi-variations.
+ * 
+ * By convention, converters that wrap an object to provide a different
+ * interface to the same underlying data structure use .asScala and .asJava
+ * extension methods, whereas converters that copy the underlying data structure
+ * use .toScala and .toJava.
+ * 
+ * In the [[javaapi]] package, the same converters can be found with a
+ * java-friendly interface that don't rely on implicit enrichments.
+ * 
+ * Additionally, this package offers [[Accumulator]]s, capable of efficiently
+ * traversing JDK Streams.
+ * 
+ * Because this package is specific to interoperating with JDK classes, its
+ * functionality makes no sense for use with scala-js and scala-native, and
+ * may not work there.
```

Comment:
Oh yeah that's just not true. Scala.js and Scala Native do implement some JDK classes, and notably the collections. They are sometimes needed because they're used by some code for performance reasons, or because we implement other Java APIs.

I suggest removing this paragraph.

Reply:
@sjrd I get a lot of linkage errors trying to use the optional converters in the jdk package, so therefore I assumed it was ok to say the functionality in the package may not work.

Is it then better to exhaustively test what does and doesn't work on scalajs and scala native and list it here?

Is the comment that it probably makes no sense to use the functionality from Scala.js and Scala Native correct, or should that also be removed?

Reply:
For `Optional` that's unsurprising, because we haven't implemented it yet. That's in the core repo, though. It's entirely possible that someone in a private codebase needed it enough that they implemented it in their codebase, and then the converters in `jdk` would link. Also if someone submits a PR with `Optional` to Scala.js, it will get merged and this will then work for everyone.

Regardless,

> Is the comment that it probably makes no sense to use the functionality from Scala.js and Scala Native correct, or should that also be removed?

It should be removed. Really, there are plenty of cases where people use that stuff in Scala.js. Typically in cross-compiling code, or code that uses a cross-compiling library that happens to expose JDK collections.